docs/en/interfaces/predefined_http.md

```diff
@@ -0,0 +1,216 @@
+# Predefined HTTP Interface {#predefined_http_interface}
+
+ClickHouse already supports specific queries through the [HTTP interface](../../interfaces/HTTP/http.md) . For example, you can write data to a table as follows:
+ 
+```bash
+$ echo '(4),(5),(6)' | curl 'http://localhost:8123/?query=INSERT%20INTO%20t%20VALUES' --data-binary @-
+```
+
+Now ClickHouse supports Predefined HTTP Interface which can help you more easy integration with third party tools. For example, the following example describes how to by Prometheus with Predefined HTTP Interface integration.
+
+> Example:
+> 
+> * First, add these configurations to the configuration file:
+> 
+> ``` xml
+> <http_handlers>
+>	<predefine_query_handler>
+>	    <url>/metrics</url>
+>	    <method>GET</method>
+>	    <queries>
+>	        <query>SELECT * FROM system.metrics LIMIT 5 FORMAT Template SETTINGS format_template_resultset = 'prometheus_template_output_format_resultset', format_template_row = 'prometheus_template_output_format_row', format_template_rows_between_delimiter = '\n'</query>
+>	    </queries>
+>	</predefine_query_handler>
+> </http_handlers>
+> ```
+> 
+> * You can now request the url directly for data in the Prometheus format:
+> 
+> ``` bash
+> curl -vvv 'http://localhost:8123/metrics'
+> *   Trying ::1...
+> * Connected to localhost (::1) port 8123 (#0)
+> > GET /metrics HTTP/1.1
+> > Host: localhost:8123
+> > User-Agent: curl/7.47.0
+> > Accept: */*
+> > 
+> < HTTP/1.1 200 OK
+> < Date: Wed, 27 Nov 2019 08:54:25 GMT
+> < Connection: Keep-Alive
+> < Content-Type: text/plain; charset=UTF-8
+> < X-ClickHouse-Server-Display-Name: i-tl62qd0o
+> < Transfer-Encoding: chunked
+> < X-ClickHouse-Query-Id: f39235f6-6ed7-488c-ae07-c7ceafb960f6
+> < Keep-Alive: timeout=3
+> < X-ClickHouse-Summary: {"read_rows":"0","read_bytes":"0","written_rows":"0","written_bytes":"0","total_rows_to_read":"0"}
+> < 
+> # HELP "Query" "Number of executing queries"
+> # TYPE "Query" counter
+> "Query" 1
+> 
+> # HELP "Merge" "Number of executing background merges"
+> # TYPE "Merge" counter
+> "Merge" 0
+> 
+> # HELP "PartMutation" "Number of mutations (ALTER DELETE/UPDATE)"
+> # TYPE "PartMutation" counter
+> "PartMutation" 0
+> 
+> # HELP "ReplicatedFetch" "Number of data parts being fetched from replica"
+> # TYPE "ReplicatedFetch" counter
+> "ReplicatedFetch" 0
+> 
+> # HELP "ReplicatedSend" "Number of data parts being sent to replicas"
+> # TYPE "ReplicatedSend" counter
+> "ReplicatedSend" 0
+> 
+> * Connection #0 to host localhost left intact
+> ```
+
+As you can see from the example, if  `<http_handlers>` is configured in the configuration file, ClickHouse will match the HTTP requests received to the predefined type in  `<http_handlers>`, then ClickHouse will execute the corresponding predefined query if the match is successful.
```

Comment:
* double spaces
* "configured in configuration file" needs to be said differently to avoid duplicate word